content/blog/2016/2016-12-16-declarative-pipeline-beta.adoc

```diff
@@ -0,0 +1,111 @@
+---
+layout: post
+title: "Declaring the Beta Release of Declarative Pipeline Syntax"
+tags:
+- pipeline
+- newfeatures
+- blueocean
+author: hrmpw
+---
+
+This week we released version 0.71 of the link:https://wiki.jenkins-ci.org/display/JENKINS/Pipeline+Model+Definition+Plugin[_Pipeline-Model-Defintion_] plugin and wanted to crown it as the official Beta version of the Declarative Pipeline syntax. Although it has been available in the update center since August, we have waited to recommend it for all of your Pipeline needs while we work out the kinks and solidify the syntax. We think this release is getting pretty close to the final version, but it is still a Beta and might change before 1.0.
+
+== What is Declarative Pipeline?
+
+All the way back at Jenkins World in September, Andrew Bayer presented a link:https://www.cloudbees.com/introducing-new-way-define-jenkins-pipelines[sneak peak] of a new syntax for constructing Pipelines. We are calling this new syntax Declarative Pipeline to differentiate it from the existing Scripted Pipeline syntax that has always been a part of Pipeline.
+
+After listening to many Jenkins users over the last year we felt that, while Scripted Pipeline provides tremendous power, flexibility, and extensibility, the learning curve for the DSL was steep for users new to either Jenkins or Pipeline. Beginning users wanting to take advantage of all the features provided by Pipeline and Jenkinsfiles were forced to learn the DSL or remain limited to the functionality provided by Freestyle jobs.
+
+Declarative Pipeline does not replace Pipeline but extends it with a pre-defined structure to let users focus entirely on the `steps` required at each `stage` without needing to worry about scripting every aspect of the pipeline. Granular flow-control is extremely powerful and Scripted Pipeline syntax will always be part of Pipeline but it's not for everyone.
+
+Declarative Pipeline enables all users to connect simple, declarative blocks that define build agents (including Docker), post build actions, environment settings, credentials and all stages that make up the pipeline. Best of all, because this Declarative syntax is part of Pipeline, all build steps and build wrappers available in Plugins or loaded from Shared Libraries are also available as `steps` in Declarative.
+
+== Example
+
+The following Pipelines should execute the same steps. The first shows the Declarative syntax and the second shows the original Scripted syntax. The Declarative syntax has a more straightforward structure that is easier to grok by users not versed in Groovy.
```

Comment:
(The two pipelines are not displayed at the same time.  The user clicks to flip between two views.)

"The following Pipelines should execute the same steps.  The first shows the Declarative syntax and the second shows the original Scripted syntax."
-> 
"Below is an example of a pipeline in Declarative syntax.  You can also switch the view to show the same pipeline in Scripted syntax."